src/Microsoft.VisualStudio.Composition.Analyzers/Utils.cs

```diff
@@ -1,26 +1,278 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-namespace Microsoft.VisualStudio.Composition.Analyzers
+namespace Microsoft.VisualStudio.Composition.Analyzers;
+
+/// <summary>
+/// Utility methods for analyzers.
+/// </summary>
+internal static class Utils
 {
-    using System;
-    using System.Collections.Generic;
-    using System.Text;
-    using Microsoft.CodeAnalysis.Diagnostics;
+    /// <summary>
+    /// Gets the URL to the help topic for a particular analyzer.
+    /// </summary>
+    /// <param name="analyzerId">The ID of the analyzer.</param>
+    /// <returns>The URL for the analyzer's documentation.</returns>
+    internal static string GetHelpLink(string analyzerId)
+    {
+        return $"https://microsoft.github.io/vs-mef/analyzers/{analyzerId}.html";
+    }
+
+    internal static bool ReferencesMefAttributes(Compilation compilation)
+    {
+        return compilation.ReferencedAssemblyNames.Any(i =>
+            string.Equals(i.Name, "System.ComponentModel.Composition", StringComparison.OrdinalIgnoreCase) ||
+            string.Equals(i.Name, "System.Composition.AttributedModel", StringComparison.OrdinalIgnoreCase));
+    }
+
+    private static readonly ImmutableArray<string> MefV1AttributeNamespace = ImmutableArray.Create("System", "ComponentModel", "Composition");
+    private static readonly ImmutableArray<string> MefV2AttributeNamespace = ImmutableArray.Create("System", "Composition");
 
     /// <summary>
-    /// Utility methods for analyzers.
+    /// Determines whether the specified attribute type is a MEF export attribute.
     /// </summary>
-    internal static class Utils
+    /// <param name="attributeType">The attribute type to check.</param>
+    /// <returns><see langword="true"/> if the attribute type is an export attribute; otherwise, <see langword="false"/>.</returns>
+    /// <remarks>
+    /// This method checks for both MEF v1 and MEF v2 export attributes, including custom attributes that derive from the base export attribute types.
+    /// </remarks>
+    internal static bool IsExportAttribute(INamedTypeSymbol? attributeType)
     {
-        /// <summary>
-        /// Gets the URL to the help topic for a particular analyzer.
-        /// </summary>
-        /// <param name="analyzerId">The ID of the analyzer.</param>
-        /// <returns>The URL for the analyzer's documentation.</returns>
-        internal static string GetHelpLink(string analyzerId)
+        if (attributeType is null)
         {
-            return $"https://microsoft.github.io/vs-mef/analyzers/{analyzerId}.html";
+            return false;
         }
+
+        // Check if it's a direct Export attribute or derived from one
+        return IsAttributeOfType(attributeType, "ExportAttribute", MefV1AttributeNamespace.AsSpan()) ||
```

Comment:
What about [`InheritedExportAttribute`](https://catalog.apireview.net/catalog/a1f8acf311f6dc566ec9d8ea45b8b969)?

Reply:
`IsAttributeOfType` walks up the base type chain, so `InheritedExportAttribute` would be considered a match.